amazon/ionbenchmark/ion_benchmark_cli.py

```diff
@@ -344,6 +344,22 @@ def reset_for_each_execution(each_option):
     return api, format_option
 
 
+def generate_ion_file_with_format(file, binary):
+    cf = open("converted_file", "wb")
+    with open(file, "br") as fp:
+        ion.dump(ion.load(fp, single_value=False), cf, binary=binary)
+    return cf.name
```

Comment:
We should do nothing if the file is already in the requested format. That way we make sure to preserve e.g. whitespace in text and symbol table boundaries in binary.

Reply:
I tried ask the tool to check file format by suffix (for ion_text) and IVM (for binary). However, suffix restriction is not supported in python-benchmark-cli yet since the tool only supports ion text and binary at the moment. 

The next step is to add a file format check mechanism and I'm going to learn from ion-java-benchmark-cli - check file header (E.g., an IVM), if it doesn't exist then fall back to suffix.